### PR TITLE
[fix][json-rpc] execute GUI.ActivateWindow asynchronous

### PR DESCRIPTION
**Pull request by Rechi**
## Description
`GUI.ActivateWindow` is a blocking call if windows is a dialog, but json-rpc requests shouldn't be blocked.
https://github.com/xbmc/xbmc/blob/9ed19a7f87e0e009b84417def5f2902ec13be2d1/xbmc/guilib/GUIWindowManager.cpp#L818-L820

## Motivation and Context
fixes https://trac.kodi.tv/ticket/16248

## How Has This Been Tested?
- `{"jsonrpc": "2.0", "method": "GUI.ActivateWindow", "id":1, "params": {"window": "shutdownmenu"}}`
- `{"jsonrpc": "2.0", "method": "Input.Down", "id":1}`

## Types of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/guilib/GUIWindowManager.cpp [hunk 1]
```diff
@@ -772,7 +772,11 @@ void CGUIWindowManager::ActivateWindow(int iWindowID, const std::vector<std::str
   {
     // make sure graphics lock is not held
     CSingleExit leaveIt(CServiceBroker::GetWinSystem()->GetGfxContext());
-    CApplicationMessenger::GetInstance().SendMsg(TMSG_GUI_ACTIVATE_WINDOW, iWindowID, swappingWindows ? 1 : 0, nullptr, "", params);
+    CGUIWindow* pWindow = GetWindow(iWindowID);
+    if (!pWindow || pWindow->IsDialog())
```
Comment (Rechi) [inline xbmc/guilib/GUIWindowManager.cpp]: I don't know if this is actually needed, maybe someone can tell me if we should block till the window (if not a dialog) is loaded.

**Comment by FernetMenta** [comment]
I don't think ActivateWindow should be asynchronous. Better make json not blocking.

**Comment by FernetMenta** [comment]
this would definitely break methods like PreviousWindow, that expect the function to be synchronous

**Comment by Rechi** [comment]
Real windows are still blocking, this change is only for dialogs and if ActivateWindow doesn't come from the main thread. So PreviousWindow should be handled correct, but I'll look into making json non blocking.

**Comment by FernetMenta** [comment]
dialogs are modal windows and PreviousWindow does also apply to dialogs. better fix the root cause which is in JSON: CBuiltins::GetInstance().Execute(cmd);

**Comment by Rechi** [comment]
@FernetMenta updated

**Review by FernetMenta — APPROVED** [review]

**Comment by DaveTBlake** [comment]
@Rechi I believe that there should be an increment in JSON API version patch number (final digit) to go with this fix. Indicates to those using JSON API that something has changed internally.

**Pull request merged**